src/transformers/models/resnet/modeling_resnet.py

```diff
@@ -194,11 +203,24 @@ def __init__(
 
         layer = ResNetBottleNeckLayer if config.layer_type == "bottleneck" else ResNetBasicLayer
 
-        self.layers = nn.Sequential(
-            # downsampling is done in the first layer with stride of 2
-            layer(in_channels, out_channels, stride=stride, activation=config.hidden_act),
-            *[layer(out_channels, out_channels, activation=config.hidden_act) for _ in range(depth - 1)],
-        )
+        if config.layer_type == "bottleneck":
+            self.layers = nn.Sequential(
+                # downsampling is done in the first layer with stride of 2
+                layer(
+                    in_channels,
+                    out_channels,
+                    stride=stride,
+                    activation=config.hidden_act,
+                    downsample_in_bottleneck=config.downsample_in_bottleneck,
+                ),
+                *[layer(out_channels, out_channels, activation=config.hidden_act) for _ in range(depth - 1)],
+            )
+        else:
+            self.layers = nn.Sequential(
+                # downsampling is done in the first layer with stride of 2
+                layer(in_channels, out_channels, stride=stride, activation=config.hidden_act),
+                *[layer(out_channels, out_channels, activation=config.hidden_act) for _ in range(depth - 1)],
+            )
```

Comment:
No as I said before the only layer that is different here is the call. Something like 
```suggestion
        if config.layer_type == "bottleneck":
                first_layer = layer(in_channels,out_channels,stride=stride,activation=config.hidden_act,downsample_in_bottleneck=config.downsample_in_bottleneck)
        else:
                first_layer =  layer(in_channels, out_channels, stride=stride, activation=config.hidden_act)
        self.layers =  nn.Sequential(first_layer,*[layer(out_channels, out_channels, activation=config.hidden_act) for _ in range(depth - 1)])
```

Reply:
something like this should be enough.

Reply:
Sure!